js/Searchbar.js

```diff
@@ -0,0 +1,89 @@
+import { SearchLayer } from './SearchLayer.js';
+
+export class Searchbar {
+  constructor($target) {
+    this.$target = $target;
+    this.render();
+    this.layer = new SearchLayer(this.$target);
+    this.focusIndex = -1;
+    this.setEvent();
+  }
+
+  template() {
+    return `<form class="search-bar__form" action="">
+              <input
+                type="text"
+                name="searchbar"
+                class="search-bar__input"
+                id="search-bar-input"
+                placeholder="검색 Amazon"
+                autocomplete="off"
+              />
+              <button class="search-bar__btn bg-orange1">
+                <img src="./assets/icons/search.svg" alt="search icon" />
+              </button>
+            </form>`;
+  }
+
+  render() {
+    this.$target.insertAdjacentHTML('beforeend', this.template());
+  }
+
+  setEvent() {
+    const searchbarInput = this.$target.querySelector('.search-bar__input');
+    const backdrop = document.querySelector('.modal__backdrop');
+
+    searchbarInput.addEventListener('click', () => {
+      const searchLayer = document.querySelector('.search-bar__layer');
+      searchLayer.classList.add('show');
+      backdrop.classList.add('show');
+      this.focusIndex = -1;
+    });
+
+    searchbarInput.addEventListener('blur', () => {
+      const searchLayer = document.querySelector('.search-bar__layer');
+      const focusEl = this.$target.querySelector(`[data-index="${this.focusIndex}"]`);
+
+      searchLayer.classList.remove('show');
+      backdrop.classList.remove('show');
+      focusEl.classList.remove('selected');
+    });
+
+    searchbarInput.addEventListener('keydown', (e) => {
+      const lists = this.$target.querySelectorAll('.search-bar__result');
+
+      if (e.key === 'ArrowDown') {
+        this.focusIndex = this.focusIndex + 1 <= lists.length - 1 ? this.focusIndex + 1 : 0;
+        const prevIndex = this.focusIndex - 1 < 0 ? lists.length - 1 : this.focusIndex - 1;
```

Comment:
코드를 볼 때 머릿속으로 계산을 해야해서 직관적이지 않을 수 있는데요!
이럴 때 부분적으로 변수화를 해주면 한결 나아집니당

```suggestion
        const isLastItem = this.focusIndex === lists.length - 1
        this.focusIndex = isLastItem ? 0 : this.focusIndex + 1
        
        const isFirstItem = this.focusIndex === 0
        const prevIndex = isFirstItem ? lists.length - 1 : this.focusIndex - 1;
```

Reply:
삼항연산자에 들어있는 의미를 알 수 없는 계산들에 변수로 이름을 붙이니 훨씬 이해가 쉬운 것 같습니다. 좋은 조언 감사합니다!